dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCWithSelf.scala

```diff
@@ -114,8 +104,68 @@ case class BinaryOutcomeDLCWithSelf(
   val sigPubKeyLose: ECPublicKey =
     Schnorr.computePubKey(messageLose, preCommittedR, oraclePubKey)
 
+  val fundingLocalPrivKey: ECPrivateKey =
+    localExtPrivKey.deriveChildPrivKey(UInt32(0)).key
+
+  val fundingRemotePrivKey: ECPrivateKey =
+    remoteExtPrivKey.deriveChildPrivKey(UInt32(0)).key
+
+  val finalLocalPrivKey: ECPrivateKey =
+    localExtPrivKey.deriveChildPrivKey(UInt32(2)).key
+
+  val finalRemotePrivKey: ECPrivateKey =
+    remoteExtPrivKey.deriveChildPrivKey(UInt32(2)).key
+
+  private val (winIndex, loseIndex) =
+    if (outcomeWin.compareTo(outcomeLose) > 0) {
```

Comment:
What's going on here? We are comparing two Strings to determine who the winner is?

Reply:
My current spec that is detailed in the PR description states that you must use the indices of the ordered set of possible outcomes (to keep things deterministic). Hence the different outcomes get a different index depending on how they compare lexicographically.